packages/dds/tree/src/feature-libraries/schema-index/formatV2.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type ObjectOptions, type Static, Type } from "@sinclair/typebox";
+
+import { SchemaCodecVersion, schemaFormatV2 } from "../../core/index.js";
+
+const noAdditionalProps: ObjectOptions = { additionalProperties: false };
+
+/**
+ * Format for encoding as json.
+ *
+ * For consistency all lists are sorted and undefined values are omitted.
+ *
+ * This chooses to use lists of named objects instead of maps:
+ * this choice is somewhat arbitrary, but avoids user data being used as object keys,
+ * which can sometimes be an issue (for example handling that for "__proto__" can require care).
+ * It also makes it simpler to determinately sort by keys.
+ */
+export const Format = Type.Object(
+	{
+		version: Type.Literal(SchemaCodecVersion.v2),
+		nodes: Type.Record(Type.String(), schemaFormatV2.TreeNodeSchemaDataFormat),
+		root: schemaFormatV2.FieldSchemaFormat,
+		metadata: schemaFormatV2.PersistedMetadataFormat,
```

Comment:
Do we have a use-case for this top level metadata? We already have metadata on the root field, and on every node, and on their fields. That seem like enough? I don't see any addition to TreeStoredSchema to allow this data to be stored on it either.